perun-client/src/Perun/Client/Multi.hs

```diff
@@ -82,6 +90,28 @@ runMultiClientWith creds action = do
     logInfo "MultiClient trace done."
     return res
 
+withChannelToken :: forall actor actors a. (HasActor actor actors) => OpenParams -> (ChannelToken -> MultiClient actors a) -> MultiClient actors a
+withChannelToken openParams action = do
+  actorStates <- gets _multiClientStateActors
+  let call = callEndpointFor @actor @actors "start" openParams
+  (cid, baseUrl) <- actionBy @actor $ do
+    cid <- gets (^. instClientId)
+    ClientEnv {baseUrl} <- gets (^. pabClientEnv)
+    return (cid, baseUrl)
+  let predf (Just (NewObservableState Null)) = False
+      predf (Just (NewObservableState _s)) = True
+      predf _ = False
+  void . liftIO . withContractSubscription baseUrl cid predf $ do
+    res <- flip evalStateT (MultiClientState actorStates) . runExceptT . unMultiClient $ call
+    case res of
+      Left err -> print err
+      Right _ -> return ()
```

Comment:
Good catch. Initially we just printed the error because we are in the `IO` monad here. We now allow embedding errors in the passed action and return them properly if they fail.